### ISSUE TITLE
Error while sending log messages to elasticsearch: message="Method may only be called on a Type for which Type.IsGenericParameter is true."

### ISSUE THREAD

**Issue by sachokFoX** [ReactiveMarkets/NLog.Targets.ElasticSearch#19]
There is error while trying to send logevent with exception to elastic. It is "known" issue in ElasticSearch.Net [SimpleJson Bug (Method may only be called on a Type for which Type.IsGenericParameter is true.)](https://github.com/elastic/elasticsearch-net/issues/2052) there is workaround with JSON.NET.

```csharp
if (logEvent.Exception != null)
    document.Add("exception", JsonConvert.DeserializeObject<ExpandoObject>(JsonConvert.SerializeObject(logEvent.Exception)));
```

Maybe add this workaround to code until main issue will be fixed. @markmcdowell @ejsmith @AlexeyRokhin ?

**Comment by markmcdowell** [comment]
I was trying to avoid the extra dependency and keeping just elastic search and nlog, hence the use of the elastic search low level client.

The serialiser can actually be swapped out at the moment but requires use of the target via code rather than config.

**Comment by markmcdowell** [comment]
Fixed by #22
